processor/src/main/java/org/mapstruct/ap/writer/IndentationCorrectingWriter.java

```diff
@@ -142,27 +162,113 @@ void onBufferFinished(StateContext context) throws IOException {
                 flush( context );
             }
 
-            private void flush(StateContext context) throws IOException {
-                if ( null != context.characters && context.currentIndex - context.lastStateChange > 0 ) {
-                    context.writer.write(
-                        context.characters,
-                        context.lastStateChange,
-                        context.currentIndex - context.lastStateChange
-                    );
+        },
 
-                    if ( DEBUG ) {
-                        System.out.print(
-                            new String(
-                                java.util.Arrays.copyOfRange(
-                                    context.characters,
-                                    context.lastStateChange,
-                                    context.currentIndex
-                                )
-                            )
-                        );
-                    }
+        /**
+         * In a String definition, Between un-escaped quotes " "
+         */
+        IN_TEXT_IN_STRING {
+
+            @Override
+            State doHandleCharacter(char c, StateContext context) {
+                switch ( c ) {
+                    case '\"':
+                        return IN_TEXT;
+                    case '\\':
+                        return IN_TEXT_IN_STRING_ESCAPE;
+                    default:
+                        return IN_TEXT_IN_STRING;
                 }
             }
+
+            /**
+             * Writes out the current text.
+             */
+            @Override
+            void onExit(StateContext context) throws IOException {
+                flush( context );
+            }
+
+            /**
+             * Writes out the current text.
+             */
+            @Override
+            void onBufferFinished(StateContext context) throws IOException {
+                flush( context );
+            }
+
+        },
+
+        /**
+         * In a String, character following an escape character (should be ignored, can also be '"' ).
+         */
+        IN_TEXT_IN_STRING_ESCAPE {
+
+            @Override
+            State doHandleCharacter(char c, StateContext context) {
+                return IN_TEXT_IN_STRING;
+            }
+
+            /**
+             * Writes out the current text.
+             */
+            @Override
+            void onExit(StateContext context) throws IOException {
+                flush( context );
+            }
+
+            /**
+             * Writes out the current text.
+             */
+            @Override
+            void onBufferFinished(StateContext context) throws IOException {
+                flush( context );
+            }
+
+        },
+
+        /**
+         * Within any text, but after a String (" ").
```

Comment:
that's indeed true.. However, only want the "onEntry" of IN_TEXT be executed once. So I cannot return to IN_TEXT. Did that, but it generates an indentation before the string-def-closing '"'. Suggestions are welcome for a better name.